File: ts-sdk/src/sdk/models/shared/storagepolicy.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum StoragePolicyTypeEnum {
    StoragePolicy = "storage_policy"
}


// StoragePolicy
/** 
 * A mini description of a Storage Policy object
**/
export class StoragePolicy extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=name" })
  name?: string;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type?: StoragePolicyTypeEnum;
}

File: sdks/python/src/boxsdk/models/shared/storagepolicy.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from boxsdk import utils

class StoragePolicyTypeEnum(str, Enum):
    STORAGE_POLICY = "storage_policy"


@dataclass_json
@dataclasses.dataclass
class StoragePolicy:
    r"""StoragePolicy
    A mini description of a Storage Policy object
    """
    
    id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    type: Optional[StoragePolicyTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    
